src/inscription.rs

```diff
@@ -44,13 +44,17 @@ impl Inscription {
       }
     }
 
-    let content_type = Media::content_type_for_extension(
-      path
-        .extension()
-        .ok_or_else(|| anyhow!("file must have extension"))?
-        .to_str()
-        .ok_or_else(|| anyhow!("unrecognized extension"))?,
-    )?;
+    let extension = path
+      .extension()
+      .ok_or_else(|| anyhow!("file must have extension"))?
+      .to_str()
+      .ok_or_else(|| anyhow!("unrecognized extension"))?;
+
+    if extension == "mp4" {
+      Media::check_mp4_codec(path)?;
+    }
+
+    let content_type = Media::content_type_for_extension(extension)?;
```

Comment:
Let's move these three checks into `Media::content_type_for_path`, e.g. extract the extension, check the codecs, and then return the `Media` value.

Reply:
Good thinking! That'll be cleaner. I'll rename the method, and do as you suggested.